piDIOComm.c

```diff
@@ -151,70 +154,69 @@ INT32U piDIOComm_Init(INT8U i8uDevice_p)
 
 INT32U piDIOComm_sendCyclicTelegram(INT8U i8uDevice_p)
 {
-	INT32U i32uRv_l = 0;
-	SIOGeneric sRequest_l;
-	SIOGeneric sResponse_l;
-	INT8U len_l, data_out[18], i, p, data_in[70];
+	static INT8U last_out[40][DIO_OUTPUT_DATA_LEN];
+
+	INT8U i, p, data_in[70];
 	INT8U i8uAddress;
+	u8 in_buf[IOPROTOCOL_MAXDATA_LENGTH];
+	u8 out_buf[DIO_OUTPUT_DATA_LEN];
+	u8 snd_buf[DIO_OUTPUT_DATA_LEN];
+	u8 snd_len;
+	u8 rcv_len;
+	u8 cmd;
 	int ret;
-	static INT8U last_out[40][18];
-#ifdef DEBUG_DEVICE_DIO
-	static INT8U last_in[40][2];
-#endif
 
-	if (RevPiDevice_getDev(i8uDevice_p)->sId.i16uFBS_OutputLength != 18) {
+	if (RevPiDevice_getDev(i8uDevice_p)->sId.i16uFBS_OutputLength !=
+	    DIO_OUTPUT_DATA_LEN) {
 		return 4;
 	}
 
-	len_l = 18;
 	i8uAddress = RevPiDevice_getDev(i8uDevice_p)->i8uAddress;
 
 	if (piDev_g.stopIO == false) {
 		rt_mutex_lock(&piDev_g.lockPI);
-		memcpy(data_out, piDev_g.ai8uPI + RevPiDevice_getDev(i8uDevice_p)->i16uOutputOffset, len_l);
+		memcpy(out_buf,
+		       piDev_g.ai8uPI + RevPiDevice_getDev(i8uDevice_p)->i16uOutputOffset,
+		       DIO_OUTPUT_DATA_LEN);
 		rt_mutex_unlock(&piDev_g.lockPI);
 	} else {
-		memset(data_out, 0, len_l);
+		memset(out_buf, 0, sizeof(out_buf));
 	}
 
 	p = 255;
-	for (i = len_l; i > 0; i--) {
-		if (data_out[i - 1] != last_out[i8uAddress][i - 1]) {
+	for (i = DIO_OUTPUT_DATA_LEN; i > 0; i--) {
+		if (out_buf[i - 1] != last_out[i8uAddress][i - 1]) {
 			p = i - 1;
 			break;
 		}
 	}
 
-	sRequest_l.uHeader.sHeaderTyp1.bitAddress = i8uAddress;
-	sRequest_l.uHeader.sHeaderTyp1.bitIoHeaderType = 0;
-	sRequest_l.uHeader.sHeaderTyp1.bitReqResp = 0;
+	snd_len = DIO_OUTPUT_DATA_LEN;
```

Comment:
`snd_len` is set in both branches of the if-else clause below, so this assignment seems unnecessary.